### ISSUE TITLE
Accuracy drop down after freezing

### ISSUE THREAD

**Issue by Sixigma** [tensorflow/tensorflow#14859]
System information

Have I written custom code (as opposed to using a stock example script provided in TensorFlow): No
OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 16.04
TensorFlow installed from (source or binary): source
TensorFlow version (use command below): 1.3.0
Python version: 2.7.12
Bazel version (if compiling from source): 0.5.4
CUDA/cuDNN version: 8.0.61
GPU model and memory: NVIDIA Corporation Device 1b06
Exact command to reproduce:

Evaluate the accuracy of ckeckpoint
```
python eval_image_classifier.py \
    --batch_size=100 \
    --checkpoint_path=.../model.ckpt-100000 \
    --dataset_dir=.../coco_tfrecord \
    --dataset_name=flowers \
    --dataset_split_name=validation \
    --model_name=mobilenet_v1 \
    --eval_dir=/tmp/eval
```

Export slim inference graph
```
python export_inference_graph.py \
--alsologtostderr \
--model_name=mobilenet_v1 \
--dataset_name=flowers \
--is_training=False \
--batch_size=1 \
--image_size=224 \
--output_file=.../mobilenet_v1.pb
```

Freeze checkpoint using slim inference graph
```
bazel-bin/tensorflow/python/tools/freeze_graph \
--input_checkpoint=.../model.ckpt-100000 \
--input_graph=.../mobilenet_v1.pb \
--input_binary=true \
--output_graph=.../frozen_graph.pb \
--output_node_names=MobilenetV1/Predictions/Reshape_1
```

### Describe the problem
I am doing some fine-tune of Mobilenet using the pretrained model.

And I follow the tutorial of slim to freeze and evaluate the performence.

I separate the validation set from train set , evaluate it using checkpoint , and do it again after freezing model , however the evaluation of  the accuracy of frozen model is far away from ckeckpoint's .

Here is some familiar issues , the situations are a little different , but I think it's referable.
#9724 
#12450

@MyHumbleSelf said " some docs should probably be adjusted " , means slim tutorial ? Do these problems remain ? 
Where might be wrong ?

Thank you !

**Comment by Sixigma** [comment]
I use the script to manually freeze model , since there is dropout and Batch normalization in Mobilenet , I set is_training = False as the suggestions ( it seems like doing the same thing as slim tutorial ) . And MovingAverage doesn't apply in model.
But the accuracy is still terrible.

**Comment by tatatodd** [comment]
@petewarden might have some ideas here.

**Comment by Sixigma** [comment]
@tatatodd Thanks for your participation !
I also notice that Mobilenet model exported directly by retrain.py is working well , but their freezing methods share the same process ( both are via "graph_util.convert_variables_to_constants" )
So I am really confused now.

**Comment by Sixigma** [comment]
I solved this by removing all Batch Normalization layers in network , followed the tutorial process and freeze model with official script , achieve the same accuracy between checkpoint and freeze_graph.pb .

And I noticed that the more different inference images comparing to training images , the loss of accuracy gets more .

Is there any other insights can be provided ?

**Comment by Sixigma** [comment]
@petewarden 
Sorry for disturbing , is there any progress on this ?

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting tensorflower` label was assigned. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Awaiting TensorFlower: It has been 14 days with no activity and the `awaiting tensorflower` label was assigned. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Awaiting TensorFlower: It has been 14 days with no activity and the `awaiting tensorflower` label was assigned. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Awaiting TensorFlower: It has been 14 days with no activity and the `awaiting tensorflower` label was assigned. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Awaiting TensorFlower: It has been 14 days with no activity and the `awaiting tensorflower` label was assigned. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging TensorFlower: It has been 14 days with no activity and the `awaiting tensorflower` label was assigned. Please update the label and/or status accordingly.
